private/bufpkg/bufplugin/bufpluginconfig/config.go

```diff
@@ -95,6 +96,11 @@ func newConfig(externalConfig ExternalConfig, options []ConfigOption) (*Config,
 	if err != nil {
 		return nil, err
 	}
+	if externalConfig.LicenseSpdxID != "" {
+		if _, ok := dataspdx.GetLicenseInfo(externalConfig.LicenseSpdxID); !ok {
```

Comment:
This does a lowercase lookup in a map and returns a `LicenseInfo` struct. I think we should canonicalize the value here before sending to the BSR and so set it to the returned `LicenseInfo.ID()` here if found.